src/Microsoft.Health.Dicom.Core/Features/Retrieve/InstanceStoreExtensions.cs

```diff
@@ -19,21 +20,37 @@ public static class InstanceStoreExtensions
 {
     public static async Task<IReadOnlyList<InstanceMetadata>> GetInstancesWithProperties(
             this IInstanceStore instanceStore,
+            IFileStore blobFileStore,
             ResourceType resourceType,
             Partition partition,
             string studyInstanceUid,
             string seriesInstanceUid,
             string sopInstanceUid,
+            bool isOriginalVersionRequested,
             CancellationToken cancellationToken)
     {
         EnsureArg.IsNotNull(instanceStore, nameof(instanceStore));
+        EnsureArg.IsNotNull(blobFileStore, nameof(blobFileStore));
 
         IReadOnlyList<InstanceMetadata> instancesToRetrieve = await instanceStore.GetInstanceIdentifierWithPropertiesAsync(partition, studyInstanceUid, seriesInstanceUid, sopInstanceUid, cancellationToken);
 
         if (!instancesToRetrieve.Any())
         {
             ThrowNotFoundException(resourceType);
         }
+        
+        // backfill each instance wit content length if none present and using external store
+        if (instancesToRetrieve.Count > 0 && instancesToRetrieve[0].InstanceProperties.FileProperties != null)
+        {
+
+            foreach (InstanceMetadata instance in instancesToRetrieve)
+            {
+                if (instance.InstanceProperties.FileProperties.ContentLength == 0)
+                {
+                    instance.InstanceProperties.FileProperties.ContentLength = (await blobFileStore.GetFilePropertiesAsync(instance.GetVersion(isOriginalVersionRequested), partition, instance.InstanceProperties.FileProperties, cancellationToken)).ContentLength;
```

Comment:
I wouldn't do it in GET call. We can utilize our job framework to populate the content length if it needs to be back filled. And let's say if I get 1000 instances we are making 1000 extra calls to blob store.